glustercli/cmd/volume-statedump.go

```diff
@@ -0,0 +1,84 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/gluster/glusterd2/pkg/api"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	flagStatedumpClient string
+	flagStatedumpQuotad bool
+
+	volumeStatedumpCmd = &cobra.Command{
+		Use:   "statedump <volname> [--quota] [--client=<host>:<pid>]",
+		Short: "Generate statedump of of a volume",
+		Long:  "Generate statedump of various processes (bricks, client, quota) of a volume. Takes statedump of all bricks by default.",
+		Args:  volumeStatedumpCmdArgs,
+		Run:   volumeStatedumpCmdRun,
+	}
+)
+
+func init() {
+	volumeStatedumpCmd.Flags().StringVar(&flagStatedumpClient, "client", "", "client process in the format <ip>:<pid>")
+	volumeStatedumpCmd.Flags().BoolVar(&flagStatedumpQuotad, "quota", false, "generate statedump of quotad process")
+	volumeCmd.AddCommand(volumeStatedumpCmd)
+}
+
+func volumeStatedumpCmdArgs(cmd *cobra.Command, args []string) error {
+
+	if len(args) != 1 {
+		return errors.New("need exactly one argument i.e name of the volume")
+	}
+
+	if _, err := cmd.Flags().GetBool("quota"); err != nil {
```

Comment:
You possibly don't need this, as pflag parsing should ensure that it is a bool, and flagStatedumpQuotad should be set correctly.